minard/burst.py

```diff
@@ -137,3 +155,138 @@ def load_bursts_search(search, start, end, offset, limit):
             total = len(results)
 
         return results, total, 0, 100
+
+def burst_get_cuts():
+    """
+    Returns a dictionary of cleaning cuts that are applied, loaded from specific couchdb document.
+    """
+    server = couchdb.Server("http://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst_cleaning"]
+
+    rows = db.view('_design/burst_cleaning/_view/cuts', descending=False, include_docs=True)
+    for row in rows:
+        doc_id = row.id
+        try:
+            result = dict(db.get(doc_id).items())
+        except KeyError:
+            app.logger.warning("Code returned KeyError searching for cut document in the couchDB.")
+
+    return result
+
+def burst_form_upload(run_number, subrun, sub, tick, summary, note, name):
+    """
+    Uploads checkbox value and note field to appropriate couchdb document - this is for SN burst review.
+    """
+    server = couchdb.Server("http://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst_cleaning"]
+
+    startkey = [run_number, subrun, sub]
+    endkey = [run_number, subrun, sub]
+    rows = db.view('_design/burst_cleaning/_view/burst_by_run', startkey=startkey, endkey=endkey, descending=False, include_docs=True)
+    for row in rows:
+        run_id = row.id
+        try:
+            doc = dict(db.get(run_id).items())
+        except KeyError:
+            app.logger.warning("Code returned KeyError searching for burst_details information in the couchDB. Run Number: %d" % run_number)
+        try:
+            doc["checked"] = tick
+            doc["summary"] = summary
+            doc["note"] = note
+            doc["reviewed_by"] = name
+            date = datetime.today().strftime("%d-%m-%Y")
+            time = datetime.today().strftime("%H:%M:%S")
+            month = calendar.month_abbr[int(date.split('-')[1])]
+            final_date = date.split('-')[0] + "-" + month + "-" + date.split('-')[2]
+            doc["review_date"] = final_date
+            doc["review_time"] = time
+            db.save(doc)
+        except KeyError:
+            app.logger.warning("Code returned KeyError appending to couchDB document. Doc ID: %s" % run_id)
+
+    return 1
+
+def translator(low_bit,high_bit,trans_map,run_type):
+    """
+    Helper function to manipulate run type bits.
+    """
+    ret = []
+    for i in range(low_bit,high_bit+1):
+        if(run_type & (1<<i)):
+            if(i-low_bit >= len(trans_map)):
+                ret.append(" SPARE (???)")
+            else:
+                ret.append(" "+trans_map[i]);
+    return ret
+
+def get_run_type(run):
```

Comment:
For me, this function is unnecessarily complicated. You could try something like: 
```
    conn = engine.connect()
    run_string = ""
    result = conn.execute("SELECT run_type FROM run_state WHERE run = %s", run)

    keys = result.keys()
    runtype = result.fetchone()[0]

    for i in range(30):
        if (runtype & (1<<i)):
            run_string += str(run_type[i])
            run_string += ", "
    run_string = run_string[:-2]

    return run_string
```
and then just remove the translator function. Then corresponding changes in the html to just have a single "run type" string rather than splitting up the bits.

Reply:
I couldn't use your suggestion straight 'out of the box' but I did mostly followed the approach